### PR TITLE
Add ability to get rates from Bakai bank

### PR DESCRIPTION
**Pull request by aloneik**
Added rates provider interface.
Added rates provider impelementation for Bakai bank. Added more informative response to user.

### CODE REVIEW

File: interfaces/currency_rates_provider.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from abc import ABC, abstractmethod
+from typing import Union, Self
+
+class CurrencyRatesProvider(ABC):
+    """Returns dict with currency code as keys and CurrencyRate instances as values"""
+    @classmethod
+    @abstractmethod
+    def get_rates(cls: Self, currency_code: Union[str, None] = None) -> dict:
```
Comment (aloneik) [inline interfaces/currency_rates_provider.py]: I think another method is needed to handle getting rates for the specific currency

File: bot.py [hunk 2]
```diff
@@ -18,15 +21,16 @@ def send_message(message):
...
-        bakai_data = bakai_response.json()
-        bakai_rub_rate = bakai_data['rates']['RUB']
+        bakai_rub_rate = BakaiRatesProvider.get_rates("RUB")["RUB"]
 
         # Send a message with ruble exchange rates from both banks
-        bot.send_message(chat_id=chat_id, text=f"Курс рубля в КГБ: {kgb_rub_rate}\nКурс рубля в Бакай: {bakai_rub_rate}")
-    except:
+        bot.send_message(
+            chat_id=chat_id,
+            text=f"Курс {bakai_rub_rate.code}-KGS в Бакай банке:\n  Наличными:\n    Покупка: {bakai_rub_rate.cash_buy}\n    Продажа: {bakai_rub_rate.cash_sell}\n  Безналичными:\n    Покупка: {bakai_rub_rate.non_cash_buy}\n    Продажа: {bakai_rub_rate.non_cash_sell}"
```
Comment (aloneik) [inline bot.py:27]: Need to add a separate method for response string building

Comment (aloneik) [reply bot.py:27]: Done in https://github.com/aloneik/kg_currency_bot/pull/4

File: interfaces/currency_rates_provider.py (hunk 1 above)
Comment (aloneik) [reply interfaces/currency_rates_provider.py]: Changed to return a tuple of CurencyRates

**Review by muchkairichka — APPROVED** [review]

**Pull request merged**